src/components/ModalForm.tsx

```diff
@@ -0,0 +1,20 @@
+import React, {useState, FormEvent, Component} from 'react'
+
+interface Props {
+    onSubmitUsername: (e: FormEvent, username: string) => void
+}
+
+export default function ModalContent(props: Props) {
+    const [username, setUsername] = useState('')
```

Comment:
because you are using `form`, when its submitted, you should be able to get the value from the `FormEvent`. because of that, the useState hook is redundant, and can be removed.

try logging it to see how you can access it

more info [here](https://react.dev/reference/react-dom/components/input#reading-the-input-values-when-submitting-a-form)